### PR TITLE
TACO-640 Include the UUID in the serialized message

### PR DESCRIPTION
**Pull request by andyjcross**
https://freshly.atlassian.net/browse/TACO-640

Important stuff here:
- Renames `MessageEnvelope` to `EventEnvelope` to standardize terminology around "events" for less confusion
- Adds the `uuid` field to `EventEnvelope` - this will give event handlers an avenue for idempotency when all else fails. 
- Adds the `StagedEvent::deserialize_event` method, which is what event subscribers will call to parse events coming in.

### CODE REVIEW

**Comment by stcho** [comment]
Hey Andy is there any manual testing steps we should worry about or is it fine just to have the specs pass for now

**Comment by andyjcross** [comment]
@stcho there isn't much to run manually at the moment, that'll happen soon

File: lib/staged_event.rb [hunk 1]
```diff
@@ -13,19 +13,54 @@
 require_relative "staged_event/railtie" if defined?(Rails)
 
 module StagedEvent
+  class Error < StandardError; end
```
Comment (stcho) [inline lib/staged_event.rb:16]: Andy why do we have to declare an Error here and not just inherit from StandardError right below?

**Review by stcho — APPROVED** [review]

File: lib/staged_event.rb (hunk 1 above)
Comment (stcho) [reply lib/staged_event.rb:16]: woops didn't mean to approve just yet, will after this question

Comment (andyjcross) [reply lib/staged_event.rb:16]: It's a good practice when designing an api.
This way, something that calls a `StagedEvent` method can rescue from _all_ errors specifically raised by `StagedEvent`, without having to rescue `StandardError` (and thus rescue _everything_ , which isn't necessarily desirable)

Comment (stcho) [reply lib/staged_event.rb:16]: Got you makes sense. The specs looks like everything is covered and it is green. Approved!

**Review by gunterja — APPROVED** [review]
This is looking awesome

**Pull request merged**